dinesykmeldte-kafka/src/main/kotlin/no/nav/syfo/sykmelding/SykmeldingService.kt

```diff
@@ -44,33 +48,66 @@ class SykmeldingService(
         }
     }
 
-    suspend fun handleSendtSykmelding(sykmeldingId: String, sykmelding: SendtSykmeldingKafkaMessage?) {
-        if (sykmelding == null) {
+    suspend fun handleSendtSykmeldingKafkaMessage(sykmeldingId: String, sykmelding: SendtSykmeldingKafkaMessage?) {
+        val existingSykmelding = sykmeldingDb.getSykmeldingInfo(sykmeldingId)
+        when (sykmelding) {
+            null -> deleteSykmelding(sykmeldingId, existingSykmelding)
+            else -> handleSendtSykmelding(sykmelding, sykmeldingId, existingSykmelding)
+        }
+        SYKMELDING_TOPIC_COUNTER.inc()
+    }
+
+    private suspend fun handleSendtSykmelding(
+        sykmelding: SendtSykmeldingKafkaMessage,
+        sykmeldingId: String,
+        existingSykmelding: SykmeldingInfo?
+    ) {
+        val sisteTom = finnSisteTom(sykmelding.sykmelding.sykmeldingsperioder)
+        if (sisteTom.isAfter(LocalDate.now().minusMonths(4))) {
+            if (sykmelding.event.arbeidsgiver == null) {
+                throw IllegalStateException("Mottatt sendt sykmelding uten arbeidsgiver, $sykmeldingId")
+            }
+            sykmeldingDb.insertOrUpdateSykmelding(toSykmeldingDbModel(sykmelding, sisteTom))
+            handleSendtSykmelding(sykmelding.kafkaMetadata.fnr)
+        } else if (existingSykmelding != null) {
```

Comment:
Bare for å være sikker på at jeg forstår denne: Hvis vi har lagret en sendt sykmelding fra før, og det kommer inn en endring på samme sykmelding som gjør at den plutselig blir eldre enn fire måneder, så skal den slettes?

Reply:
ja riktig